client/services/states/states.service.js

```diff
@@ -1,59 +1,179 @@
-angular.module('probedock.states').factory('states', function($rootScope, $state, $stateParams, $timeout) {
+angular.module('probedock.states').factory('states', function($rootScope, $transitions) {
 
-  var onStateCallbacks = [];
+  var currentState = null,
+      onStateChangeStartCallbacks = [],
+      onStateChangeSuccessCallbacks = [];
 
-  $rootScope.$on('$stateChangeSuccess', checkState);
+  var service = {
 
-  function checkState(event, toState, toParams, fromState, fromParams) {
```

Comment:
The comments on the onStateChnage* functions are really useful.

I would just suggest to add a little precision about the fact a scope is mandatory and in case we want to add a callback that will persist during the whole lifecyle of the app. In other terms, explanation on when to user $rootScope.